packages/utils/src/color/Color.ts

```diff
@@ -0,0 +1,305 @@
+    toString(): string
+    {
+        const hexString = this._int.toString(16);
+
+        return `#${'000000'.substring(0, 6 - hexString.length) + hexString}`;
+    }
+
+    /**
+     * Convert to a hexidecimal string with alpha.
+     * @example
+     * import { utils } from 'pixi.js';
+     * new utils.Color('white').toHexString(); // returns "#ffffffff"
+     */
+    toHexString(): string
+    {
+        const alphaValue = Math.round(this._components[3] * 255);
+        const alphaString = alphaValue.toString(16);
+
+        return this.toString() + '00'.substring(0, 2 - alphaString.length) + alphaString;
+    }
+
+    /**
+     * Set alpha, suitable for chaining.
+     * @param alpha
+     */
+    setAlpha(alpha: number): this
+    {
+        this._components[3] = alpha;
+
+        return this;
+    }
+
+    /** Get the alpha amount */
+    toAlpha(): number
+    {
+        return this._components[3];
+    }
+
+    /**
+     * Rounds the specified color according to the step.
+     * @param step - Number of steps which will be used as a cap when rounding colors
+     */
+    round(step: number): this
+    {
+        const [r, g, b] = this._components;
+
+        this._components.set([
+            Math.min(255, (r / step) * step),
+            Math.min(255, (g / step) * step),
+            Math.min(255, (b / step) * step),
+        ]);
+
+        return this;
+    }
+
+    /**
+     * Converts color to an [R, G, B, A] array of normalized floats (numbers from 0.0 to 1.0).
+     * @example
+     * import { utils } from 'pixi.js';
+     * new utils.Color('white').toArray(); // returns [1, 1, 1, 1]
+     * @param {number[]|Float32Array} [out] - Output array
+     */
+    toArray<T extends (number[] | Float32Array) = number[]>(out?: T): T
+    {
+        out = out ?? [] as T;
+        const [r, g, b, a] = this._components;
+
+        out[0] = r;
+        out[1] = g;
+        out[2] = b;
+        out[3] = a;
+
+        return out;
+    }
+
+    /** Normalize the input value into rgba */
+    private normalize(): void
+    {
+        let value = this._value;
+        let components: number[];
+
+        if ((Array.isArray(value) || value instanceof Float32Array) && value.length >= 3)
+        {
+            // Handle floats or uints
+            const normalizeValue = (v: number) => (v > 1 ? v / 255 : v);
+
+            components = [
+                normalizeValue(value[0]),
+                normalizeValue(value[1]),
+                normalizeValue(value[2]),
+                value[3] ?? 1.0,
+            ];
+        }
+        else if (typeof value === 'string' || typeof value === 'object')
+        {
+            if (typeof value === 'string')
+            {
+                if (value.startsWith('0x'))
+                {
+                    value = value.slice(2);
+                }
+                if (Color.HEX_PATTERN.test(value) && !value.startsWith('#'))
+                {
+                    value = `#${value}`;
+                }
```

Comment:
```suggestion
                if (value.startsWith('0x'))
                {
                    value = `#${value.slice(2)}`;
                }
                else if (Color.HEX_PATTERN.test(value))
                {
                    value = `#${value}`;
                }
```

Original code may accept string like `0x#ffffff`, which I think should be rejected.

Also because the `#` is removed from the regex, we don't need to check it any more.

Reply:
Did some more cleanup on this